jenkins/Jenkinsfile

```diff
@@ -0,0 +1,148 @@
+                            echo \$NXPASS | podman login docker-public-nexus.meteoswiss.ch -u \$NXUSER --password-stdin
+
+                            test -d ctx && rm -rf ctx
+                            mkdir ctx
+
+                            cp docker/Dockerfile docker/spack.yaml docker/entrypoint.sh docker/upload_s3.py docker/IGBP_int1.dat docker/sandbox_generator.py docker/genconf.yml ctx/
+                            podman build --format docker --pull --build-arg TOKEN_FLEXPART=$GITHUB_ACCESS_TOKEN --build-arg container_registry=\$CR_ALL --build-arg spack_tag=\$SPACK_TAG -t \$IMAGE_INTERN ctx
+                            podman push \$IMAGE_INTERN
+                            podman tag \$IMAGE_INTERN \$IMAGE_PUBLIC \$IMAGE_ECR
+                            if [ "${PUBLIC_RELEASE}" = 1 ]; then podman push \$IMAGE_PUBLIC; fi
+                        '''
+                    }
+                    if (params.PUSH_IMAGE_TO_ECR) {
+                        withCredentials([usernamePassword(credentialsId: 'aws-icon-sandbox',
+                                                    passwordVariable: 'AWS_SECRET_ACCESS_KEY',
+                                                    usernameVariable: 'AWS_ACCESS_KEY_ID')]) {
+                            echo "---- PUBLISH IMAGE TO ECR ----"
+                            sh """
+                            #!/bin/bash
+
+                            if test -f /etc/ssl/certs/ca-certificates.crt; then
+                                export AWS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+                            else
+                                export AWS_CA_BUNDLE=/etc/ssl/certs/ca-bundle.crt
+                            fi
+                            aws ecr get-login-password --region eu-central-2 | podman login --username AWS --password-stdin --cert-dir /etc/ssl/certs \$REPO_ECR
+
+                            podman push --cert-dir /etc/ssl/certs \$IMAGE_ECR
+
+                            """
+                        }
+                    } 
+                }
+            } 
+            post {
+                cleanup {
+                    script {
+                        // Remove the context directory and images
+                        sh """
+                        rm -rf ctx
+                        podman image rm -f \$IMAGE_INTERN
+                        podman image rm -f \$IMAGE_PUBLIC
+                        podman image rm -f \$IMAGE_ECR
+                        podman logout docker-intern-nexus.meteoswiss.ch || true
+                        podman logout docker-public-nexus.meteoswiss.ch || true
+                        """
+                        if (params.PUSH_IMAGE_TO_ECR) {
+                            sh "podman logout ${env.REPO_ECR} || true"
+                        }
+                    }
+                }
+            }
+        }
+        stage('Test container') {
```

Comment:
Build and test should be done before Deploy stage . I would copy the layout here: https://github.com/C2SM-RCM/flexpart/blob/main/Jenkinsfile